src/main/java/org/spongepowered/api/item/inventory/menu/ClickHandler.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory.menu;
+
+import org.spongepowered.api.item.inventory.Container;
+
+@FunctionalInterface
+public interface ClickHandler {
+
+    /**
+     * Handler for inventory click callbacks except without slot.
+     *
+     * @param container the container
+     * @param clickType the type of click
+     *
+     * @return false to prevent the action
+     */
+    boolean handle(Container container, ClickType clickType);
```

Comment:
Whereas Container nets you a lot of information, it looses a lot of the context of the click. Any argument for Cause/EventContext in this case? These aren't real events I'm aware but I feel like valuable information would be lost.

Same argument I made to @dualspiral for moving CommandExecutor over to having Cause/etc.

Reply:
Agree with `Cause` and `EventContext` thing, it may be very useful.

Reply:
I'll play both sides here:

We don't explicitly need to provide the cause, because the handler a) can be on either side of the thread management, b) still retrieved by `Sponge.getCauseStackManager().getCurrentCause()` if needed, c) Forces our implementation to generate the Cause even if it's not used (since this is not event based, we gain some free performance if the plugin is able to cope without cause usage at all).